internal/baggage/restriction_manager.go

```diff
@@ -18,32 +18,51 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package jaeger
+package baggage
 
 const (
 	defaultMaxValueLength = 2048
 )
 
-// baggageRestrictionManager keeps track of valid baggage keys and their size restrictions.
-type baggageRestrictionManager interface {
-	// setBaggage sets the baggage key:value on a span and the associated logs.
-	setBaggage(span *Span, key, value string)
+type Restriction struct {
+	valid          bool
+	maxValueLength int
 }
 
-// defaultBaggageRestrictionManager allows any baggage key.
-type defaultBaggageRestrictionManager struct {
-	setter baggageSetter
+func NewRestriction(valid bool, maxValueLength int) *Restriction {
```

Comment:
not a fan of this, pollutes namespace and if we ever need to add extra restrictions, can't easily update the signature without a breaking change.